packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -660,21 +660,16 @@ class _AndroidMotionEventConverter {
   }
 
   AndroidPointerProperties propertiesFor(PointerEvent event, int pointerId) {
-    int toolType = AndroidPointerProperties.kToolTypeUnknown;
-    switch (event.kind) {
-      case PointerDeviceKind.touch:
-      case PointerDeviceKind.trackpad:
-        toolType = AndroidPointerProperties.kToolTypeFinger;
-      case PointerDeviceKind.mouse:
-        toolType = AndroidPointerProperties.kToolTypeMouse;
-      case PointerDeviceKind.stylus:
-        toolType = AndroidPointerProperties.kToolTypeStylus;
-      case PointerDeviceKind.invertedStylus:
-        toolType = AndroidPointerProperties.kToolTypeEraser;
-      case PointerDeviceKind.unknown:
-        toolType = AndroidPointerProperties.kToolTypeUnknown;
-    }
-    return AndroidPointerProperties(id: pointerId, toolType: toolType);
+    return AndroidPointerProperties(
+      id: pointerId,
+      toolType: switch (event.kind) {
+        PointerDeviceKind.touch || PointerDeviceKind.trackpad => AndroidPointerProperties.kToolTypeFinger,
+        PointerDeviceKind.mouse => AndroidPointerProperties.kToolTypeMouse,
+        PointerDeviceKind.stylus => AndroidPointerProperties.kToolTypeStylus,
+        PointerDeviceKind.invertedStylus => AndroidPointerProperties.kToolTypeEraser,
+        PointerDeviceKind.unknown || _ => AndroidPointerProperties.kToolTypeUnknown,
```

Comment:
You have introduced default cases in many locations, which violates https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values and makes the codebase harder to maintain. If that is required in order to use switch expressions in these scenarios, that's a non-starter.

Reply:
Instead of reading through the enum, I noticed that `toolType` was initialized to `AndroidPointerProperties.kToolTypeUnknown`, meaning if no cases matched then that would be its value.

But it looks like all `PointerDeviceKind` values are being covered here, so it didn't have to be initialized (and thus the switch expression default case can be removed).

Thanks for letting me know about this rule! I'm going to read through the style guide again to make sure I haven't made other mistakes; hopefully soon I'll have a flawless PR :)

Reply:
> Instead of reading through the enum, I noticed that `toolType` was initialized to `AndroidPointerProperties.kToolTypeUnknown`, meaning if no cases matched then that would be its value.

Even if adding a default case keeps the same runtime behavior, it changes the behavior of an important lint for future changes, which is why the style rule exists.

Reply:
Also, using `_` makes it the default behavior for future `enum` values, which may silently introduce unwanted behavior. By being explicit we statically guarantee that we evaluated every possibility, and when we introduce new values to the `enum` we won't have any surprise.

Reply:
@mateusfccp  I'm not sure why you say "Also" there; that's the same thing my comments above are about.

Reply:
If this is the case, I misunderstood what you said.